Automattic/wp-calypso
# Storage Add Ons: Implement Plans 2023 Features Grid Functionality

Related to https://github.com/Automattic/wp-calypso/pull/79041#discussion_r1279224891
Issue https://github.com/Automattic/martech/issues/2025
Tracking https://github.com/Automattic/martech/issues/2023

## Proposed Changes

* Update use-add-on hook to return monthly and yearly add-on costs
* Implement use add-on hook in plan-features-main component
* Add storageAddOns data to gridPlans
* Consume storage add-on data in the storage dropdown component, header pricing component, and billing timeframe component

## GIFs
![2023-08-31 16 10 43](https://github.com/Automattic/wp-calypso/assets/5414230/157f43f0-6f5c-458c-b822-93fb922cf61d)

## Needs to be fixed
* Mobile view layout and styling

## Will Follow Up in Future PR
* Comparison grid storage add on dropdown width is too narrow for tablet resolutions ( longer currency values are wonky )
* Make the storage add-on in checkout cart a dropdown ( to allow user to switch between storage add-ons )
* Fix plans upgrades button in `/plans` page. They currently don't add the storage add-on into the checkout cart properly
* Add storage add-on dropdown to the `/plans` spotlight card

## Testing Instructions



* Smoke test signup flows with business or ecommerce plans:
  - `/start?flags=plans%2Fupgradeable-storage`
  - `/start/domain?flags=plans%2Fupgradeable-storage` 
  - `/start/plans-import?flags=plans%2Fupgradeable-storage`
  - `/start/onboarding-pm?flags=plans%2Fupgradeable-storage`
* Smoke test stepper flows with business or ecommerce plans:
  - `setup/onboarding-media?flags=plans%2Fupgradeable-storage`
  - `/setup/link-in-bio-tld?flags=plans%2Fupgradeable-storage`
* Verify that the storage add-on dropdown only renders for business and ecommerce plans
* Verify that the dropdown is only rendered for annual plans ( not monthly )
* Verify that the header pricing and billing timeframe changes appropriately when storage add-ons are selected
* Verify that the checkout cart is loaded with the selected storage add-on when a plan is selected

## Pre-merge Checklist
